src/test/java/hudson/plugins/git/extensions/impl/PruneStaleTagTest.java

```diff
@@ -0,0 +1,237 @@
+        GitClient remoteClient = initRepository(listener, remoteRepo);
+
+        // clone remote repository to workspace
+        GitClient localClient =  cloneRepository(remoteRepo);
+
+        GitSCM scm = new GitSCM(localClient.getRemoteUrl("origin"));
+        PruneStaleTag extension = new PruneStaleTag();
+
+        // new tags should not be fetched
+        String tagName = "tag";
+        String tagComment = "tag comment";
+        remoteClient.tag(tagName, tagComment);
+        extension.decorateFetchCommand(scm, run, localClient, listener, null);
+        Assert.assertFalse("new tags should not be fetched", localClient.tagExists(tagName));
+    }
+
+    /*
+     * Local tag       | Remote tag
+     * -------------------------------
+     * exists - same   | exists - same
+     */
+    @Test
+    public void verify_that_local_tag_is_not_pruned_when_exist_on_remote() throws Exception {
+        File remoteRepo = fileRule.newFolder("remote");
+
+        // create a remote repository with one tag
+        GitClient remoteClient = initRepository(listener, remoteRepo);
+        String tagName = "tag";
+        String tagComment = "tag comment";
+        remoteClient.tag(tagName, tagComment);
+
+        // clone remote repository to workspace
+        GitClient localClient =  cloneRepository(remoteRepo);
+
+        GitSCM scm = new GitSCM(localClient.getRemoteUrl("origin"));
+        PruneStaleTag extension = new PruneStaleTag();
+
+        extension.decorateFetchCommand(scm, run, localClient, listener, null);
+        Assert.assertTrue("local tags must not be pruned when exists on remote", localClient.tagExists(tagName));
+    }
+
+    /*
+     * Local tag       | Remote tag
+     * -------------------------------
+     * exists          | not exists
+     */
+    @Test
+    public void verify_that_local_tag_is_pruned_when_not_exist_on_remote() throws Exception {
+        File remoteRepo = fileRule.newFolder("remote");
+
+        // create a remote repository with one tag
+        GitClient remoteClient = initRepository(listener, remoteRepo);
+        String tagName = "tag";
+        String tagComment = "tag comment";
+        remoteClient.tag(tagName, tagComment);
+
+        // clone remote repository to workspace
+        GitClient localClient =  cloneRepository(remoteRepo);
+
+        GitSCM scm = new GitSCM(localClient.getRemoteUrl("origin"));
+        PruneStaleTag extension = new PruneStaleTag();
+
+        // remove tag on remote, tag remains on local cloned repository
+        remoteClient.deleteTag(tagName);
+        extension.decorateFetchCommand(scm, run, localClient, listener, null);
+        Assert.assertFalse("local tag has not been pruned", localClient.tagExists(tagName));
+    }
+
+    @Test
+    public void verify_fetch_do_not_prune_local_branches() throws Exception {
+        File remoteRepo = fileRule.newFolder("remote");
+
+        // create a remote repository with one tag
+        initRepository(listener, remoteRepo);
```

Comment:
Where is created the tag? Or do you mean _without tags_?
```suggestion
        // create a remote repository without any tag
        initRepository(listener, remoteRepo);
```